aiosmtpd/smtp.py

```diff
@@ -89,10 +89,13 @@ def _client_connected_cb(self, reader, writer):
         self._writer = writer
 
     def connection_lost(self, exc):
```

Comment:
Do we even need `connection_lost()` any more?  Specifically, does it add anything over what the super class's method provides?  I'll note that test coverage on your branch never even touches this method.  Why not just remove it?